### PR TITLE
adding curl command in the attachement of each api + checking tests

### PR DESCRIPTION
**Pull request by tekeuange23**
Main contributor: @olivierodo  

## Task: Curl Command
- adding curl command in the attachement of each api && checking tests

### CODE REVIEW

File: src/restqapi/lib/api/index.js [hunk 1]
```diff
@@ -28,10 +28,65 @@ module.exports = function (options) {
...
         error: error && error.message
       }
+    },
+    getCurl: function () {
+      let { protocol, hostname, pathname, method, searchParams, headers, json } = this.request.getOptions()
+      const { bodyBackup } = this.request
+      const curlCommand = []
+
+      // console.log({ ...bodyBackup, xyz: 1111111111 })
+      // console.log(this.request.getOptions())
```
Comment (olivierodo) [inline src/restqapi/lib/api/index.js:43]: Console.log? 😉

File: src/restqapi/lib/api/index.js [hunk 2]
```diff
@@ -28,10 +28,65 @@ module.exports = function (options) {
...
+      if (searchParams) {
+        curlCommand.push('?')
+        const params = []
+        for (const key in searchParams) {
+          params.push(key + '=' + searchParams[key])
+        }
+        curlCommand.push(params.join('&'))
+      }
+
+      return curlCommand.join('')
```
Comment (olivierodo) [inline src/restqapi/lib/api/index.js:89]: Can you remove the spaces on each `curlCommand.push` and add it in the `curlCommand.join`

**Review by olivierodo — CHANGES_REQUESTED** [review]
Overall it's good. 
Just small changes are required

File: src/restqapi/lib/api/index.js (hunk 1 above)
Comment (tekeuange23) [reply src/restqapi/lib/api/index.js]: OK

File: src/restqapi/lib/api/index.js (hunk 2 above)
Comment (tekeuange23) [reply src/restqapi/lib/api/index.js]: Ok let see

Comment (tekeuange23) [reply src/restqapi/lib/api/index.js]: done!!!

**Review by olivierodo — APPROVED** [review]

**Pull request merged**